app/models/compute_resources/foreman/model/libvirt.rb

```diff
@@ -237,6 +237,31 @@ def normalize_vm_attrs(vm_attrs)
       normalized
     end
 
+    def self.terminate_connection(url = nil)
+      if url
+        key = "libvirt_connection_#{url}"
+        conn = Thread.current[key]
+      else
+        Thread.current.keys.each do |a_key|
```

Comment:
Why aren't we terminating them all? if you have multiple libvirt Compute Resources, this will kill just the first, right?